qiskit/transpiler/passes/layout/csp_layout.py

```diff
@@ -100,10 +101,15 @@ def __init__(
     def run(self, dag):
         """run the layout method"""
         qubits = dag.qubits
-        cxs = set()
+        needed_connections = set()
 
         for gate in dag.two_qubit_ops():
-            cxs.add((qubits.index(gate.qargs[0]), qubits.index(gate.qargs[1])))
+            needed_connections.add((qubits.index(gate.qargs[0]), qubits.index(gate.qargs[1])))
+
+        for gate in dag.multi_qubit_ops():
```

Comment:
`two_qubits_ops` and `multi_qubits_ops` do very similar things:
```python
        ops = []
        for node in self.op_nodes(include_directives=False):
            if len(node.qargs) >= X:
                ops.append(node)
        return ops
```
So probably it is better to call `op_nodes` and avoid the double iteration on the structure.

Reply:
thx. change made. iterating directly over op_nodes now.